### PR TITLE
Improve build performance by removing the "proptypes-index" generation

### PR DESCRIPTION
**Pull request by DanReyLop**
I toyed a bit with the build, and I realized that `Happypack` isn't actually improving anything for the server build speed, so I removed it. At worse it's the same, and at best it's maybe 1 second faster now.

Also, the task that was taking the longest (`30s` on my machine) was `generate-proptypes-index`. I optimized it (I got it down to 25 seconds), and then I realized it wasn't being used at all, so I removed it entirely and now it takes exactly 0 seconds to complete, a speed improvement of 100%. I've added instructions in https://github.com/Automattic/wp-calypso/pull/10342 if we want to add it again.

To test, just run `npm run build` or `npm start`, check that `devdocs` works as usual and notice that the `server` step of the build is a bit faster.

### CODE REVIEW

**Comment by samouri** [comment]
what is the propTypes index for?

**Comment by DanReyLop** [comment]
It was going to be added to each component's `devdocs` page. See https://github.com/Automattic/wp-calypso/pull/10342. But since that PR was abandoned (for now), then it's just dead code taking time on the build.

File: package.json [hunk 1]
```diff
@@ -215,7 +212,7 @@
     "lint:css": "stylelint \"client/**/*.scss\" --syntax scss",
     "lint:js": "npm run -s install-if-deps-outdated && eslint-eslines .",
     "lint:mixedindent": "mixedindentlint --ignore-comments \"client/**/*.scss\" \"assets/**/*.scss\" \"**/*.js\" \"**/*.jsx\" \"!build/**\" \"!node_modules/**\" \"!public/**\" \"!client/config/index.js\"",
-    "minify": "./bin/minify",
+    "minify": "node bin/minify.js",
```
Comment (samouri) [inline package.json]: is there any impact to this change?

**Comment by samouri** [comment]
> But since that PR was abandoned

I can't believe it wasn't removed right there and then 🤕. Thanks for this @DanReyLop!
I wonder if there are other parts of the build process that don't need to be done during dev build (generate proptypes usage, css splitting, code splitting, etc)

File: package.json (hunk 1 above)
Comment (DanReyLop) [reply package.json]: Only that now it works on Windows. Windows doesn't interpret the `#! /usr/bin/node` header in a file, so you have to be explicit and invoke it with `node ./path/to/file`.

**Comment by DanReyLop** [comment]
> I wonder if there are other parts of the build process that don't need to be done during dev build (generate proptypes usage, css splitting, code splitting, etc)

I didn't find any other big offenders. All the dev `build` tasks are done in parallel, and now the slowest of them is `build-server` (`webpack`), and that's unavoidable.

**Comment by samouri** [comment]
Tested:
- [ ] local dev build
- [ ] docker build

**Comment by samouri** [comment]
@DanReyLop 

Can you add back in HappyPack? I have a feeling you forgot to test without the babe-cache.  In order to test without the babel-cache you can run:
1. `rm -rf build/.babel-server-cache`
2. `npm run build-server`

Build times with HappyPack
1. without cache: 18s
2. with cache: 11s

Build times without HappyPack:
1. without cache 45s
2. with cache: 11s

So HappyPack has a negligible effect when the cache exists because babel-loaded doesn't need to do any CPU intensive work anymore.  It helps drastically when there is no cache though.

**Comment by DanReyLop** [comment]
My bad, I didn't know `babel-server-cache` was a thing. I should have assumed HappyPack was there for a reason :) . I rolled back that commit.

**Pull request merged**